content/blog/gatsby-blog/index.md

```diff
@@ -0,0 +1,323 @@
+
+It should look like this
+
+```json
+"deploy": "gatsby build --prefix-paths && gh-pages -d public -b master"
+```
+
+- `--prefix-paths` is only useful if you are going to use the default github domain: `username.github.io/reponame`. Remove it if you are making your site accessible from some `domain.com`
+- `gh-pages` is a helper node package that you need to install: `npm install --dev gh-pages`
+- `-b master` specifies to `gh-pages` package that you want to deploy to the `master` branch
+
+
+### 3 Adjust your `pathPrefix`
+
+IMPORTANT! only do this if you are using the default github domain. If you are using a custom domain,
+like `myblog.com` you should probably leave it unset, or if you want your blog to live in a custom path
+of your custom domain like `myblog.com/custompath` then adjust accordingly.
+
+
+Go to `gatsby-config.js` and set the top level property 
+
+```javascript
+pathPrefix: reponame
+```
+
+
+
+### 4 deploy!
+
+```bash
+
+# make sure you commit and push your changes to develop
+git commit ...
+git push origin develop
+
+# deploy!
+npm run deploy
+```
+
+Now if you go to `username.github.io/reponame` you should be able to navigate your site!
+
+Congratulations! your site is live!
+
+
+## [optional] Buy a domain and use it
+
+We used `godaddy` but we have no affiliations with them and there are plenty of
+domain name providers. Choose the right one for you, buy your domain and then adjust DNS accordingly.
+
+
+### 1 Setup your DNS
+
+Most of the domain providers will let you set up the DNS records for your site. We are going to use
+that to redirect to your github page, set up the following records in your DNS managing tool.
+
+Read the official [instructions](https://help.github.com/en/github/working-with-github-pages/about-custom-domains-and-github-pages).
+
+But here is the gist:
+
+- **for apex domain**: create an 4 `A` records, each with one of these IPs
+```bash
+185.199.108.153
+185.199.109.153
+185.199.110.153
+185.199.111.153
+```
+- **for `www.`**: create a `CNAME` record with a key of `www` and a value of `username.github.io`
+
+You may have both without a problem
+
+
+
+### 2 Setup you CNAME in github
+
+If you are using `npm run deploy` to deploy you app then go to your repo's settings,
+scroll to the bottom and add the custom domain in the **Custom Domain** input.
+You should add your naked or apex domain, without the `www`.
+
+If you are using github actions (more on this later) then you need to add
+this exact file to your source code in the `develop` branch,
+it should be named exactly `CNAME` and it should contain your domain as the only line of text.
+
+CNAME
+```bash
+myblog.com
+```
+
+
+Depending on your domain provider these changes in the DNS records might take
+from a couple of minutes to a couple of days (in our case it was pretty fast with godaddy).
+
+Checkout the `dig` linux/unix command to diagnose DNS updates (probably will do a blog post about this)
```

Comment:
I think the parenthesis sentence would benefit if it said "separate blog post"/"another blog post", but it's 100% a stylistic choice.

Reply:
hm... let me have this one